bin/backport-wp-commit.sh

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env bash
+
+set -e
+
+wp_changeset="$1"
+if [[ ! "$wp_changeset" =~ ^[0-9]+$ ]]; then
+	echo "Usage: $0 WP_CHANGESET_NUMBER [--current-branch]"
+	echo
+	echo "  WP_CHANGESET_NUMBER  The SVN changeset number to port.  If this change depends"
+	echo "                       on other changes, make sure they have already been ported."
+	echo "  --current-branch     Apply the commit directly to the current branch instead of"
+	echo "                       creating a new branch."
+	echo
+	exit 1
+fi
+
+create_branch=yes
+if [ "$2" = "--current-branch" ]; then
+	create_branch=no
+fi
+
+# Sanity check: make sure we have a .git directory, and at least one remote
+# pointing to a repository named ClassicPress
+if [ ! -d .git ] || ! git remote -v | grep -q '\b/ClassicPress\b'; then
+	echo "ERROR: Call this script from within your ClassicPress repository"
+	exit 1
+fi
+
+# Make sure there are no modified files in the local repository
+change_type=
+if ! git diff --exit-code --quiet; then
+	change_type="Modified file(s)"
+elif ! git diff --cached --exit-code --quiet; then
+	change_type="Staged file(s)"
+fi
+if [ ! -z "$change_type" ]; then
+	git status
+	echo
+	echo "ERROR: $change_type detected"
+	echo "ERROR: You must start this script from a clean working tree!"
+	exit 1
+fi
+
+cmd() {
```

Comment:
I don't understand the purpose of this command, can you explain it?

Reply:
Check for modified files before doing anything. If you have edited files in your working tree, or changes staged but not committed, then it's not a good idea to run this script.